packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -150,6 +150,13 @@ export const render = async (req, res, next) => {
         })
     }
 
+    // Some application extensions need to be serialized because they have asynchronous state
+    const serializedExtensions = Object.fromEntries(
+        applicationExtensions
+            .map((extension) => [extension.getName(), {routes: extension.serializeAsyncRoutes()}])
+            .filter(([, value]) => Array.isArray(value.routes) && value.routes.length > 0)
```

Comment:
Instead of filtering after running the serialize, why now only serialize the extensions that should be serialized? ... in the jsdoc comment above you say "some" .. maybe that is a good indicator to use the some() function on the application extensions array.